crates/matrix-sdk-base/src/client.rs

```diff
@@ -980,10 +968,12 @@ impl BaseClient {
                 }
 
                 changes
-                    .members
+                    .state
                     .entry(room_id.to_owned())
                     .or_default()
-                    .insert(member.state_key().to_owned(), raw_event.clone().cast());
+                    .entry(member.event_type())
+                    .or_default()
+                    .insert(member.state_key().to_owned().into(), raw_event.clone().cast());
```

Comment:
Is this `.to_owned()` necessary?

Reply:
Yes (well it could be a `.clone()`), `Into<String>` is not implemented for `&OwnedUserId`.

Reply:
How about `.to_string()` instead of `.to_owned().into()`? I think it's more obvious, and shouldn't be any less efficient.